src/data/equiv/basic.lean

```diff
@@ -1625,6 +1625,15 @@ begin
   rw swap_apply_of_ne_of_ne hi hj,
 end
 
+lemma swap_inj_apply [decidable_eq β] {f : α → β} (hf : function.injective f) (x y z : α) :
+  swap (f x) (f y) (f z) = f (swap x y z) :=
```

Comment:
Is this lemma statement at all similar to [`function.update_comp_eq_of_injective`](https://leanprover-community.github.io/mathlib_docs/data/set/function.html#function.update_comp_eq_of_injective)? If so, can we try and make the name more similar, and/or use it in the proof?

Reply:
That lemma also suggest we might want to write this
```
swap (f x) (f y) ∘ f = f ∘ swap x y
```

Reply:
I know often the fully applied statement is preferred over ones that involve `∘`. What's the guidance on preferring one or the other?

Reply:
I don't know - what you have is probably fine. It's tempting to call this `swap_comp_eq_of_injective` for consistency though.

Reply:
I think we can have both.